src/TopDownProteomics/ProForma/ProFormaParser.cs

```diff
@@ -291,34 +312,35 @@ private void ProcessTag(string tag, int? startIndex, int index, ref IList<ProFor
 
                 if (!string.IsNullOrEmpty(group))
                 {
-                    if (tagGroups == null) tagGroups = new Dictionary<string, ProFormaTagGroup>();
+                    tagGroups ??= new Dictionary<string, ProFormaTagGroup>();
 
                     if (!tagGroups.ContainsKey(group))
                     {
-                        tagGroups.Add(group, new ProFormaTagGroupChangingValue(group, key, evidence, new List<ProFormaMembershipDescriptor>()));
+                        tagGroups.Add(group, new ProFormaTagGroup(group, key, evidence, string.Empty, new List<ProFormaMembershipDescriptor>()));
                     }
 
                     var currentGroup = tagGroups[group];
 
                     // Fix up name of TagGroup
-                    if (!string.IsNullOrEmpty(value) && currentGroup is ProFormaTagGroupChangingValue x)
+                    if (!string.IsNullOrEmpty(value) && currentGroup is ProFormaTagGroup x)
                     {
                         // Only allow the value of the group to be set once
                         if (!string.IsNullOrEmpty(x.Value))
                             throw new ProFormaParseException($"You may only set the value of the group {group} once.");
 
-                        x.ValueFlux = value;
+                        x.Value = value;
                         x.Key = key;
                         x.EvidenceType = evidence;
+                        x.PreferredLocation = currentGroup.Members.Count;
```

Comment:
I'm not sure I get this. Shouldn't this be the one that has the more descriptive tag, rather than the last one? If I'm understanding this correctly, this makes the test pass, but it doesn't really satisfy the concept of the notation. 
<img width="848" alt="image" src="https://user-images.githubusercontent.com/16342951/219508306-81f626d4-8528-4a8e-92f2-e91b6204e8a0.png">